docs/developer/first-transform.md

```diff
@@ -0,0 +1,148 @@
+As we mentioned elsewhere, [setting up a generator](/developer/setting-up-generator/) is a great way to establish a repository template from which all of your newer projects can derive. There's one more powerful function you can apply when generating a project, and that's a _code transform_.
+
+A code transform allows you to change any of the files in a seed repository at the time of a project's generation. You can use it to do anything from changing the copyright year in source code header licenses to manipulating the AST of the files entirely.
+
+In this example, we'll apply a code transform that does two things to a Node project: first, it'll update the README.md to note the project's name, and second, it'll also update the package.json file to use the project's name, too.
+
+## Prerequisites
+
+You should already have [a blank SDM](/developer/sdm/#creating-an-sdm-project), though you can also use an existing one if you'd like.
+
+We'll be using the [atomist-seeds/express-es6-rest-api](https://github.com/atomist-seeds/express-es6-rest-api) repository as our seed. Please don't use this seed for your own Express projects, as this is just a demo app (and likely full of outdated packages!).
+
+## Registering a generator
+
+Since we've already gone through this before, we'll simply provide the generic generator boilerplate to get started.  Open up your SDM, and add a new file called `lib/node_generator.ts`. Paste the following lines to get started with your generator:
+
+```typescript
+import { GitHubRepoRef } from "@atomist/automation-client";
+import { GeneratorRegistration } from "@atomist/sdm";
+
+export const NodeExpressGenerator: GeneratorRegistration = {
+  name: "Node Express Project",
+  intent: "generate-node-express-project",
+  startingPoint: GitHubRepoRef.from({
+      owner: "atomist-seeds",
+      repo: "express-es6-rest-api",
+  }),
+  transform: []
+};
+```
+
```

Comment:
should we try the generator first? i really want to try it first. (we have to register it with the SDM)